libethcore/CommonJS.cpp

```diff
@@ -22,16 +22,27 @@
  */
 
 #include "CommonJS.h"
+#include "ICAP.h"
 
 namespace dev
 {
 
-Address toAddress(std::string const& _sn)
+Address jsToAddress(std::string const& _s)
 {
-	if (_sn.size() == 40)
-		return Address(fromHex(_sn));
-	else
-		return Address();
+	try
+	{
+		eth::ICAP i = eth::ICAP::decoded(_s);
+		return i.direct();
+	}
+	catch (eth::InvalidICAP&) {}
+	try
+	{
+		auto b = fromHex(_s.substr(0, 2) == "0x" ? _s.substr(2) : _s, WhenError::Throw);
```

Comment:
should we add more warnings, like:
* check if address is already part of the state (just a warning, it is ok, but may be not intented)
* warning for `0x00..00` address
* check length
...